### PR TITLE
add kafka eventbus spec

### PR DESCRIPTION
**Pull request by premadk**
Signed-off-by: Prema devi Kuppuswamy <premadk@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

File: pkg/apis/eventbus/v1alpha1/kafka_eventbus.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	corev1 "k8s.io/api/core/v1"
+)
+
+// KafkaBus holds the KafkaBus EventBus information
+type KafkaBus struct {
+	// Exotic holds an exotic Kafka config
+	Exotic *KafkaConfig `json:"exotic,omitempty" protobuf:"bytes,1,opt,name=exotic"`
+}
+
+type KfConsumerGroup struct {
```
Comment (dfarr) [inline pkg/apis/eventbus/v1alpha1/kafka_eventbus.go]: Is it possible to name this `KafkaConsumerGroup`? Also can we define this below `KafkaConfig` as this is the order they are used.

File: manifests/base/controller-manager/controller-config.yaml [hunk 2]
```diff
@@ -57,3 +57,10 @@ data:
           metricsExporterImage: natsio/prometheus-nats-exporter:0.9.1
           configReloaderImage: natsio/nats-server-config-reloader:0.7.0
           startCommand: /nats-server
+      kafka:
+        streamConfig: |
+          # The default properties of the streams to be created in this Kafka service
```
Comment (dfarr) [inline manifests/base/controller-manager/controller-config.yaml]:
```suggestion
          # Default Kafka settings, can be overridden by EventBus Kafka specs
```

Comment (dfarr) [reply manifests/base/controller-manager/controller-config.yaml]: Just changing this to align with the jetstream comment

File: manifests/install.yaml [hunk 3]
```diff
@@ -351,6 +351,13 @@ data:
           metricsExporterImage: natsio/prometheus-nats-exporter:0.9.1
           configReloaderImage: natsio/nats-server-config-reloader:0.7.0
           startCommand: /nats-server
+      kafka:
+        streamConfig: |
+          # The default properties of the streams to be created in this Kafka service
```
Comment (dfarr) [inline manifests/install.yaml]:
```suggestion
          # Default Kafka settings, can be overridden by EventBus Kafka specs
```

File: manifests/namespace-install.yaml [hunk 4]
```diff
@@ -271,6 +271,13 @@ data:
           metricsExporterImage: natsio/prometheus-nats-exporter:0.9.1
           configReloaderImage: natsio/nats-server-config-reloader:0.7.0
           startCommand: /nats-server
+      kafka:
+        streamConfig: |
+          # The default properties of the streams to be created in this Kafka service
```
Comment (dfarr) [inline manifests/namespace-install.yaml]:
```suggestion
          # Default Kafka settings, can be overridden by EventBus Kafka specs
```

File: pkg/apis/eventbus/v1alpha1/kafka_eventbus.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	// Rebalance strategy can be one of: sticky, roundrobin, range. Range is the default.
+	// +optional
+	RebalanceStrategy string `json:"rebalanceStrategy,omitempty" protobuf:"bytes,2,opt,name=rebalanceStrategy"`
+	// When starting up a new group do we want to start from the oldest event (true) or the newest event (false), defaults to false
+	// +optional
+	StartOldest bool `json:"startOldest,omitempty" default:"false" protobuf:"bytes,3,opt,name=startOldest"`
+}
+
+type KafkaConfig struct {
+	// Kafka url
```
Comment (dfarr) [inline pkg/apis/eventbus/v1alpha1/kafka_eventbus.go]:
```suggestion
	// URL to kafka cluster, multiple URLs separated by comma
```

**Review by dfarr — CHANGES_REQUESTED** [review]

File: manifests/base/controller-manager/controller-config.yaml (hunk 2 above)
Comment (premadk) [reply manifests/base/controller-manager/controller-config.yaml]: that's for the settings of jetstream.. jetstream's stream config have this one "# The default properties of the streams to be created in this JetStream service" thats already in sync.. 
but i still feel it need to be rephrased

Comment (dfarr) [reply manifests/base/controller-manager/controller-config.yaml]: Whoops, updated my comment

**Review by dfarr — APPROVED** [review]
:+1:

**Pull request merged**